cli/cmd/install.go

```diff
@@ -33,156 +30,99 @@ This command provides the Kubernetes configs necessary to install the Buoyant
 Cloud Agent.
 
 If an agent is not already present on the current cluster, this command
-redirects the user to Buoyant Cloud to set up a new agent. Once the new agent is
-set up, this command will output the agent manifest.
+will provide you with a manifest that should perform auto registration of
+the agent with the name you have specified.
 
 If an agent is already present, this command retrieves an updated manifest from
-Buoyant Cloud and outputs it.`,
-		Example: `  # Default install.
+Buoyant Cloud and outputs it.
+
+Note that this command required CLIENT_ID and CLIENT_SECRET env vars to be set.
+To get a CLIENT_ID and CLIENT_SECRET, log in to https://buoyant.cloud.`,
+		Example: `  # Default install (no agent on cluster).
+  linkerd buoyant install --agent-name=my-new-agent | kubectl apply -f -
+
+  # Obtain a manifest for an agent that already exists on your cluster
   linkerd buoyant install | kubectl apply -f -
 
   # Install onto a specific cluster
   linkerd buoyant --context test-cluster install | kubectl --context test-cluster apply -f -`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			client, err := k8s.New(cfg.kubeconfig, cfg.kubecontext, cfg.bcloudServer)
+			client, err := k8s.New(installCfg.kubeconfig, cfg.kubecontext)
 			if err != nil {
 				return err
 			}
 
-			return install(cmd.Context(), cfg, client, browser.OpenURL)
+			// get clientID and clientSecret from either k8s or env vars
+			var clientID, clientSecret string
+
+			secret, err := client.Secret(cmd.Context())
+			if err == nil && secret != nil {
+				clientID = string(secret.Data["client_id"])
+				clientSecret = string(secret.Data["client_secret"])
+			}
+
+			// prefer env vars
+			if os.Getenv("CLIENT_ID") != "" {
+				clientID = os.Getenv("CLIENT_ID")
+			}
+			if os.Getenv("CLIENT_SECRET") != "" {
+				clientSecret = os.Getenv("CLIENT_SECRET")
+			}
+
+			if clientID == "" {
+				return errors.New("install command requires setting CLIENT_ID")
+			}
+			if clientSecret == "" {
+				return errors.New("install command requires setting CLIENT_SECRET")
+			}
+
+			apiClient := bcloudapi.New(clientID, clientSecret, installCfg.bcloudAPI, !installCfg.noTLS)
```

Comment:
I think this needs to be `installCfg.noTLS` instead of `!installCfg.noTLS`.

I tried running the agent install command from this branch against an AKS cluster, which should require TLS, and it failed with this error:

```
$ ./bin/go-run cli install --context cluster1     
Error: failed to retrieve agent manifest from bcloud server for agent identifier bcloudapi.AgentID XXXXXXXXXXX: Get "http://api.buoyant.cloud:443/agent.yaml?agent_id=XXXXXXXXXXX": oauth2: cannot fetch token: 400 Bad Request
Response: <html>
<head><title>400 The plain HTTP request was sent to HTTPS port</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<center>The plain HTTP request was sent to HTTPS port</center>
<hr><center>nginx</center>
</body>
</html>
```

> The plain HTTP request was sent to HTTPS port

`./bin/go-run cli install --context cluster1 --no-tls` fixes it, but the `--no-tls` flag shouldn't be there.